supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/PipelinesApiInitializationActor.scala

```diff
@@ -131,6 +139,61 @@ class PipelinesApiInitializationActor(pipelinesParams: PipelinesApiInitializatio
     } yield encrypted
   }
 
+  lazy val vpcNetworkAndSubnetworkProjectLabels: VpcAndSubnetworkProjectLabelValues = {
+    def googleProject(descriptor: BackendWorkflowDescriptor): String = {
+      descriptor.workflowOptions.getOrElse(WorkflowOptionKeys.GoogleProject, pipelinesParams.jesConfiguration.papiAttributes.project)
+    }
+
+    def projectMetadataRequest(vpcConfig: VirtualPrivateCloudConfiguration): IO[HttpRequest] = {
+      IO {
+        val credentials = vpcConfig.auth.credentials(workflowOptions.get(_).get, List(CloudLifeSciencesScopes.CLOUD_PLATFORM))
+
+        val httpCredentialsAdapter = new HttpCredentialsAdapter(credentials)
+        val cloudResourceManagerBuilder = new CloudResourceManager
+        .Builder(GoogleAuthMode.httpTransport, GoogleAuthMode.jsonFactory, httpCredentialsAdapter)
+          .setApplicationName(pipelinesConfiguration.googleConfig.applicationName)
+          .build()
+
+        val project = cloudResourceManagerBuilder.projects().get(googleProject(workflowDescriptor))
+
+        project.buildHttpRequest()
+      }
+    }
+
+    def projectMetadataResponseToLabels(httpResponse: HttpResponse): IO[ProjectLabels] = {
+      implicit val googleProjectMetadataLabelDecoder: Decoder[ProjectLabels] = deriveDecoder
+      IO.fromEither(decode[ProjectLabels](httpResponse.parseAsString())).handleErrorWith {
+        e => IO.raiseError(new RuntimeException(s"Failed to parse labels from project metadata response from Google Cloud Resource Manager API. " +
+          s"${ExceptionUtils.getMessage(e)}", e))
+      }
+    }
+
+    def networkLabelsFromProjectLabels(vpcConfig: VirtualPrivateCloudConfiguration, projectLabels: ProjectLabels): VpcAndSubnetworkProjectLabelValues = {
+      val networkLabelOption = projectLabels.labels.collectFirst {
+        case (labelName, labelValue) if labelName.equals(vpcConfig.name) => labelValue
+      }
+      val subnetworkLabelOption = vpcConfig.subnetwork.flatMap { s =>
+        projectLabels.labels.collectFirst {
+          case (labelName, labelValue) if labelName.equals(s) => labelValue
+        }
+      }
+
+      VpcAndSubnetworkProjectLabelValues(networkLabelOption, subnetworkLabelOption)
+    }
+
+    def fetchVpcLabelsFromProjectMetadata(vpcConfig: VirtualPrivateCloudConfiguration): IO[VpcAndSubnetworkProjectLabelValues] = {
+      for {
+        projectMetadataResponse <- projectMetadataRequest(vpcConfig).map(_.executeAsync().get())
+        projectLabels <- projectMetadataResponseToLabels(projectMetadataResponse)
+      } yield networkLabelsFromProjectLabels(vpcConfig, projectLabels)
+    }
+
+    pipelinesConfiguration.papiAttributes.virtualPrivateCloudConfiguration match {
+      case None => VpcAndSubnetworkProjectLabelValues(None, None)
+      case Some(vpcConfig) => fetchVpcLabelsFromProjectMetadata(vpcConfig).unsafeRunSync()
```

Comment:
Not safe to use `unsafeRunAsync` and it's going to start throwing exceptions in weird and unexpected places if this goes wrong.

Since the call site for `vpcNetworkAndSubnetworkProjectLabels` below is already a `for`-comp, we should make sure any errors in this function are handed by the same logic which handles any other invalid `for`-comp result.